pkg/registry/extensions/thirdpartyresourcedata/codec.go

```diff
@@ -384,8 +384,8 @@ func (t *thirdPartyResourceDataDecoder) Decode(data []byte, gvk *schema.GroupVer
 		if gvk == nil {
 			return nil, nil, runtime.NewMissingKindErr(string(data))
 		}
-		mapObj["kind"] = gvk.Kind
-		actual.Kind = gvk.Kind
+		mapObj["kind"] = t.kind
```

Comment:
do you mean we need to keep setting `gvk.Kind` for `mapObj["kind"]`? as I think `gvk.Kind` will always hold the value of "ThirdPartyResource", so that if user provides an empty kind field, value "ThirdPartyResource" will be saved in data and act as a placeholder for `kind` field?